src/lib/FlightTasks/tasks/ManualPositionSmoothVel/FlightTaskManualPositionSmoothVel.cpp

```diff
@@ -157,27 +151,38 @@ void FlightTaskManualPositionSmoothVel::_updateSetpoints()
 		}
 	}
 
-	Vector3f pos_sp_smooth;
-	Vector3f accel_sp_smooth;
+	if (!_position_lock_xy_active) {
+		_smoothing[0].setCurrentPosition(_position(0));
+		_smoothing[1].setCurrentPosition(_position(1));
+	}
+	if (!_position_lock_z_active) {
+		_smoothing[2].setCurrentPosition(_position(2));
+	}
 
+	Vector3f pos_sp_smooth;
 	for (int i = 0; i < 3; ++i) {
-		if (!_position_lock_xy_active) {
-			_smoothing[i].setCurrentPosition(_position(i));
-		}
 
-		_smoothing[i].integrate(accel_sp_smooth(i), _vel_sp_smooth(i), pos_sp_smooth(i));
+		_smoothing[i].integrate(_acceleration_setpoint(i), _vel_sp_smooth(i), pos_sp_smooth(i));
 		_velocity_setpoint(i) = _vel_sp_smooth(i); // Feedforward
+		_jerk_setpoint(i) = _smoothing[i].getCurrentJerk();
 	}
 
 	// Check for position lock transition
```

Comment:
This is not only executed on a lock transition. Is that intended?

Reply:
Yes, the "position lock" slows down and stops automatically when the velocity reaches 0. This way, the position setpoint is consistent with its derivatives.